neo/Network/P2P/TaskManager.cs

```diff
@@ -234,7 +234,8 @@ private void RequestTasks(TaskSession session)
                         break;
                     }
                 }
-                session.RemoteNode.Tell(Message.Create("getblocks", GetBlocksPayload.Create(hash)));
+                if (!globalTasks.ContainsKey(hash))
```

Comment:
Hmm if the hash is in globalTasks it could be getting data for the block, but this is asking for inventory and since this doesn’t give hash stop it will potentially get newer block inventory, so this could still be useful to leave the way it is actually.

Reply:
I think that you did not understand what I mean.

Currently there is an error, check it out because the else case depicted in the example shows that.

Reply:
If that is the case you mention we should act before calling the method.

Reply:
There are 2 errors, Jeff.

1. Ask for a hash that is already in GlobalTask
2. Reply with 500 blocks anyway...

Both of this should be fixed, a part of fulfilling the network.

The plausible possibilities is:
1. Ask for blocks in windows ( as I proposed in a previous PR, with a loop)

There is no sense to ask for a block and another node reply 500.
That is why we have Blocker Header made in batches of 2000.

Reply:
Hi, Jeff, I think that there are probably 2 errors:

1. Ask for a hash that is already in GlobalTask
2. Always reply with 500 blocks.

Both of this should be fixed because they help to fulfill the network and respond with unnecessary payloads.

The plausible possibility I thought was:
1. Ask for blocks in windows ( as I proposed in a previous PR, with a loop #486)

There is no sense to ask for a block and another node reply 500.
That is why we have Blocker Header made in batches of 2000.

Reply:
It looks like you are right again, @jsolman.

It kind of always reply with 500 blocks (which is not desirable from my perspective).
In this sense, I still think that the plausible possibility I thought was to ask for blocks in windows ( as I proposed in a previous PR, with a loop #486).